### PR TITLE
refactor pd.

### PR DESCRIPTION
**Pull request by siddontang**
+ Add PutRegion to report region status to pd regularly. 
+ Remove AskChangePeer, Pd can handle it in PutRegion directly. 
+ Remove pdjob, no need job now. 

@ngaut @qiuyesuifeng @disksing @tiancaiamao

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

File: proto/pdpb.proto [hunk 1]
```diff
@@ -16,12 +17,11 @@ enum CommandType {
     AllocId             = 4;
     GetStore            = 5;
     PutStore            = 6;
-    DeleteMeta          = 7;
```
Comment (qiuyesuifeng) [inline proto/pdpb.proto]: What is `DeleteMeta` used for before？

File: proto/pdpb.proto [hunk 2]
```diff
@@ -146,9 +163,9 @@ message Request {
     optional AllocIdRequest alloc_id                        = 6;
     optional GetStoreRequest get_store                      = 7;
     optional PutStoreRequest put_store                      = 8;
-    optional AskChangePeerRequest ask_change_peer           = 9;
-    optional AskSplitRequest ask_split                      = 10;
-    optional GetRegionRequest get_region                    = 11;
+    optional AskSplitRequest ask_split                      = 9;
+    optional GetRegionRequest get_region                    = 10;
+    optional GetRegionRequest put_region                    = 11;
```
Comment (qiuyesuifeng) [inline proto/pdpb.proto]: PutRegionRequest?

File: proto/pdpb.proto [hunk 3]
```diff
@@ -162,9 +179,9 @@ message Response {
     optional AllocIdResponse alloc_id                       = 6;
     optional GetStoreResponse get_store                     = 7;
     optional PutStoreResponse put_store                     = 8;
-    optional AskChangePeerResponse ask_change_peer          = 9;
-    optional AskSplitResponse ask_split                     = 10;
-    optional GetRegionResponse get_region                   = 11;
+    optional AskSplitResponse ask_split                     = 9;
+    optional GetRegionResponse get_region                   = 10;
+    optional GetRegionResponse put_region                   = 11;
```
Comment (qiuyesuifeng) [inline proto/pdpb.proto]: PutRegionResponse?

File: proto/pdpb.proto (hunk 1 above)
Comment (siddontang) [inline proto/pdpb.proto]: no used before

File: proto/pdpb.proto (hunk 2 above)
Comment (siddontang) [inline proto/pdpb.proto]: oh, yep, my fault.

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by disksing** [comment]
LGTM

**Pull request merged**